src/index.ts

```diff
@@ -0,0 +1,41 @@
+import express, { Express, request, Request, Response } from 'express';
+import { Book } from './book';
+
+const app = express()
+const books: Book[] = []
+
+app.use(express.json());
+
+app.post('/book', (req: Request, res: Response) => {
+    books.push(req.body);
+    res.status(200).json(books);
+});
+
+app.get('/book/:searchByVal', (req: Request, res: Response) => {
+    const searchByVal = req.params.searchByVal;
```

Comment:
Cool synthetic sugar 
```suggestion
    const { searchByVal } = req.params;
```